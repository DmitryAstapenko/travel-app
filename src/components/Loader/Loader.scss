.loader {
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;

  & > .spinner-border {
    width: 5rem;
    height: 5rem;
    border: .25em solid;
    border-right: .25em solid transparent;
  }
}
