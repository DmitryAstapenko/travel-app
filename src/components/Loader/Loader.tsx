import * as React from "react";
import { Spinner } from "react-bootstrap";
import "./Loader.scss";

const Loader: React.FunctionComponent = () => {
  return (
    <div className="loader">
      <Spinner animation="border" />
    </div>
  );
};

export default Loader;
