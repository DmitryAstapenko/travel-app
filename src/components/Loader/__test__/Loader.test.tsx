import React from "react";
import Loader from "../Loader";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

const setUp = () => shallow(<Loader />);

describe("should render loader component", () => {
  let component;
  beforeEach(() => {
    component = setUp();
  });
  it("should render loader component", () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
