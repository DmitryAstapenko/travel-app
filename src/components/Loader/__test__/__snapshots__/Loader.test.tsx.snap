// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render loader component should render loader component 1`] = `
<div
  className="loader"
>
  <Spinner
    animation="border"
  />
</div>
`;
